normandy/recipes/api/serializers.py

```diff
@@ -119,3 +119,26 @@ class Meta:
             'date_created',
             'recipe',
         ]
+
+
+class SignatureSerializer(serializers.ModelSerializer):
+    timestamp = serializers.DateTimeField(read_only=True)
+    signature = serializers.ReadOnlyField()
+    x5u = serializers.ReadOnlyField()
+    public_key = serializers.ReadOnlyField()
+
+    class Meta:
+        model = Signature
+        fields = ['timestamp', 'signature', 'x5u', 'public_key']
+
+
+class SignedRecipeSerializer(serializers.ModelSerializer):
+    signature = SignatureSerializer()
+    recipe = serializers.SerializerMethodField()
```

Comment:
I tried that first. The problem is that this serializer is serializing a `Recipe`, and needs to pass itself to the nested serializer. If this line were just `recipe = RecipeSerializer()`, it would try and look for `self.recipe`, which doesn't exist. I also couldn't figure out anything I could supply as the `source` for the serializer to make it point to itself either.